app/src/main/java/io/bankingofthings/iot/Finn.kt

```diff
@@ -286,32 +297,88 @@ class Finn(
             .apply { disposables.add(this) }
     }
 
+    /**
+     * Trigger action.
+     */
     @Throws(
         ActionFrequencyNotFoundError::class,
         ActionFrequencyTimeNotPassedError::class,
-        ActionTriggerFailedError::class
+        ActionTriggerFailedError::class,
+        ActionTriggerFailedAlternativeIdRequired::class,
+        ActionNotActivatedError::class
     )
+    fun triggerAction(actionID: String, alternativeID: String? = null): Completable {
+        if (multiPair && alternativeID == null) {
+            throw ActionTriggerFailedAlternativeIdRequired()
+        } else {
+
+            return if (networkManager.hasNetworkConnection()) {
+                createCompositeSendAction(actionID, alternativeID)
+                    .onErrorResumeNext {
+                        when (it::class) {
+                            HttpException::class -> {
+                                if ((it as HttpException).code() == 400) {
+                                    Completable.error(ActionNotActivatedError())
+                                } else {
+                                    Completable.error(it)
+                                }
+                            }
+                            else -> Completable.error(it)
+                        }
```

Comment:
it can be simplified like this:
```
    when {
            it is HttpException && it.code() == 400 ->  Completable.error(ActionNotActivatedError())
            else -> Completable.error(it)
        }
```